### PR TITLE
redefinition of sidb_nm_position

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

Redefinition of ``sidb_nm_position``. The return value was previously in meters and was changed to nanometers for consistency.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: include/fiction/technology/charge_distribution_surface.hpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ class charge_distribution_surface<Lyt, false> : public Lyt
         else
         {
             strg->phys_params = params;
-            this->initialize_distance_matrix();
+            this->initialize_distance_nm_matrix();
```
Comment (marcelwa) [inline include/fiction/technology/charge_distribution_surface.hpp:228]: Similarly here

File: include/fiction/technology/charge_distribution_surface.hpp [hunk 2]
```diff
@@ -409,9 +409,9 @@ class charge_distribution_surface<Lyt, false> : public Lyt
      * @param index2 The second index.
      * @return The distance index between `index1` and `index2` (indices correspond to unique SiDBs).
      */
-    [[nodiscard]] double get_distance_by_indices(const uint64_t index1, const uint64_t index2) const noexcept
+    [[nodiscard]] double get_distance_nm_by_indices(const uint64_t index1, const uint64_t index2) const noexcept
```
Comment (marcelwa) [inline include/fiction/technology/charge_distribution_surface.hpp:412]: Similar here

**Review by marcelwa — CHANGES_REQUESTED** [review]
Many thanks for the changes. Looks all good code-wise. I just have a couple grammar-related points.

File: include/fiction/technology/charge_distribution_surface.hpp [hunk 3]
```diff
@@ -390,14 +390,14 @@ class charge_distribution_surface<Lyt, false> : public Lyt
      *
      *  @param c1 the first cell to compare.
      *  @param c2 the second cell to compare.
-     *  @return a constexpr double representing the distance between the two cells.
+     *  @return a constexpr double representing the distance in nm between the two cells.
      */
-    [[nodiscard]] double get_distance_between_cells(const typename Lyt::cell& c1,
-                                                    const typename Lyt::cell& c2) const noexcept
+    [[nodiscard]] double get_distance_nm_between_cells(const typename Lyt::cell& c1,
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/technology/charge_distribution_surface.hpp:395]: good idea!

**Review by marcelwa — APPROVED** [review]

**Pull request merged**